test/data/javascript/tests.js

```diff
@@ -316,6 +319,242 @@ exports.test_data = {
+                ]
+            }, {
+                comment: 'catch-all, includes brackets and other various code',
+                input: inputlib.operator_position.catch_all,
+                output: [
+                    'var d = 1;',
+                    'if (a === b',
+                    '    && c) {',
+                    '    d = (c * everything',
+                    '            / something_else)',
+                    '        % b;',
+                    '    e',
+                    '        += d;',
+                    '',
+                    '} else if (!(complex && simple)',
+                    '    || (emotion && emotion.name === "happy")) {',
+                    '    cryTearsOfJoy(many',
+                    '        || anOcean',
+                    '        || aRiver);',
+                    '}'
+                ]
+            }
+        ]
+    }, {
+        name: "operator_position option - set to 'preserve_newline'",
+        options: [{
+            name: "operator_position", value: "'preserve_newline'"
+        }],
+        tests: [
+            {
+                comment: 'comprehensive, various newlines',
+                input: inputlib.operator_position.comprehensive,
+                output: [
+                    'var res = a + b',
+                    '    - c /',
+                    '    d * e',
+                    '    %',
+                    '    f;',
+                    'var res = g & h',
+                    '    | i ^',
+                    '    j;',
+                    'var res = (k &&',
+                    '        l',
+                    '        || m) ?',
+                    '    n',
+                    '    : o;',
+                    'var res = p',
+                    '    >> q <<',
+                    '    r',
+                    '    >>> s;',
+                    'var res = t',
+                    '',
+                    '    === u !== v',
+                    '    !=',
+                    '    w',
+                    '    == x >=',
+                    '    y <= z > aa <',
+                    '    ab;',
+                    'ac +',
+                    '    -ad'
+                ]
+            }, {
+                comment: 'colon special case',
+                input: inputlib.operator_position.colon_special_case,
+                output: [
+                    'var a = {',
+                    '    b: bval,',
+                    '    c: cval,',
+                    '    d: dval',
+                    '};',
+                    'var e = f ? g',
+                    '    : h;',
+                    'var i = j ? k :',
+                    '    l;'
+                ]
+            }, {
+                comment: 'catch-all, includes brackets and other various code',
+                unchanged: inputlib.operator_position.catch_all
+            }
+        ]
+    }, {
+        name: "reset operator_position",
+        options: [{ name: "operator_position", value: "'before_newline'" }]
```

Comment:
Alright fixed the three comments - let me know when you're done and I'll push a fourth draft.  When we're finished I'll rebase them all into a single commit